README.md

```diff
@@ -53,7 +59,7 @@ In one sentence, explain why.
 
 ## A1.2
 
-«replace this with your answer»
+O(n). One way the parser could work is like checking if a string is a palindrome by setting a pointer at the front and back and incremenet the one at the front and decrement the one at the back until they land on the same character since it needs to have an odd length, checking whether the two are pointing at the same character each time, which would require a checking of n number of tiles. (Also, the parser could first check whether n is even or odd because if it is even, then it would know that it is not valid.)
```

Comment:
All true, but you're really described a specific algorithm for this particular problem, rather than an actual parse of the grammar itself.

The answer is still O(n), but the parse would likely involve pushing tokens onto a stack, then checking if the remaining tokens are the same as popping those tokens back off. This would be a O(n^2) speed algorithm, but O(n) in memory.

# -1